cli/flox/src/commands/environment.rs

```diff
@@ -1198,6 +1199,11 @@ impl Push {
                 To rename your environment: 'flox edit --name <new name>'
                 To pull and manually re-apply your changes: 'flox delete && flox pull -r {owner}/{name}'
             "}.into(),
+            ManagedEnvironmentError::Build(_) => formatdoc! {"
+                ❌  Unable to push environment with compatibility errors on your system ({system}).
```

Comment:
nonblocking product nit @ghudgins @jennymahmoudi: I don't think I would know as a user what `compatibility errors` means. And because we're suppressing the underlying error, I don't actually know what to do next. I can't start with `flox edit`; I'd actually have to `flox activate` to see the error, and then do `flox activate && flox edit && flox push`.

Also `compatibility errors` makes it sound like it's a system specific issue, but this is the same error I get if I do:
```
flox init
# edit the manifest and misspell a package name
flox push
```

Reply:
I agree with the concerns reg. "compatibility".
The way its implemented now its ensuring that the environment _can build_ on the current system.
I don't know if it becomes much clearer with the error present

(`-v` will show it FYI)

Reply:
the AC calls to show both errors 
```
And When the env fails to build
Then I see the failure from the build
And I see a message about push
```

Reply:
ok to add the error in a somewhat raw form now (and later/soon use the prettyfied version)?

Reply:
yes - we'll redo the error in a future ticket #602

Reply:
> Also compatibility errors makes it sound like it's a system specific issue, but this is the same error I get if I do:
```
 flox init 
# edit the manifest and misspell a package name
flox push
```

Is this still a valid concern? 
Can we have separate error messages for `flox push` due to system compatibility issues vs misspelled package names?

Reply:
Yeah it's still a concern. At the moment I don't think it would be very easy to have a different message, but we could come back to that when we re-work our errors. Until then I think it would be simplest to say `Unable to push environment that doesn't build` or whatever word we're currently using for build

Reply:
Ok I think it's fine to continue to use build since that's already being propagated through in the other error message.

`Unable to push environment with build errors.
`